### ISSUE TITLE
/sys/bus/w1/devices/ 1-Wire SysBus shows no devices

### ISSUE THREAD

**Issue by Napoleon-BlownApart** [thecode/ha-onewire-sysbus#46]
From the CLI, I can easily see both of my thermometers in /sys/bus/w1/devices/ and view the temperature they are reporting; so it's not a device issue.  But the integration in HA shows nothing; per image attached:  
![One-Wire_Integration_n_Config](https://user-images.githubusercontent.com/6958677/193742961-ecd1a487-e6f4-4156-89b1-e68e89637767.png)

My devices are not discovered automatically (as they were in the past),  there are no options to add entities/devices, and the documentation for the integration does not mention how to add them to HA.

Anyone know what to do at this point?

**Comment by thecode** [comment]
Please add more information, your HA core version, CLI output & sensor type/model

**Comment by Napoleon-BlownApart** [comment]
HA 2022.9.7
2 x DS18B20 Thermometers connected to the same Raspberry Pi that is running HA.
CLI output:
```
user@host:/sys/bus/w1/devices $ ll
total 0
drwxr-xr-x 2 root root 0 Oct  4 12:27 .
drwxr-xr-x 4 root root 0 Oct  4 12:27 ..
lrwxrwxrwx 1 root root 0 Oct  4 16:06 28-00000588c73f -> ../../../devices/w1_bus_master1/28-00000588c73f
lrwxrwxrwx 1 root root 0 Oct  4 16:06 28-00000589da65 -> ../../../devices/w1_bus_master1/28-00000589da65
lrwxrwxrwx 1 root root 0 Oct  4 13:17 w1_bus_master1 -> ../../../devices/w1_bus_master1
user@host:/sys/bus/w1/devices $ cat 28-00000588c73f/w1_slave
59 01 4b 46 7f ff 07 10 a2 : crc=a2 YES
59 01 4b 46 7f ff 07 10 a2 t=21562
user@host:/sys/bus/w1/devices $
```

**Comment by thecode** [comment]
> HA 2022.9.7 running in a Docker Container on Raspbian Desktop. 2 x DS18B20 Thermometers connected to the same Raspberry Pi that is running HA. CLI output:
> 
> ```
> user@host:/sys/bus/w1/devices $ ll
> total 0
> drwxr-xr-x 2 root root 0 Oct  4 12:27 .
> drwxr-xr-x 4 root root 0 Oct  4 12:27 ..
> lrwxrwxrwx 1 root root 0 Oct  4 16:06 28-00000588c73f -> ../../../devices/w1_bus_master1/28-00000588c73f
> lrwxrwxrwx 1 root root 0 Oct  4 16:06 28-00000589da65 -> ../../../devices/w1_bus_master1/28-00000589da65
> lrwxrwxrwx 1 root root 0 Oct  4 13:17 w1_bus_master1 -> ../../../devices/w1_bus_master1
> user@host:/sys/bus/w1/devices $ cat 28-00000588c73f/w1_slave
> 59 01 4b 46 7f ff 07 10 a2 : crc=a2 YES
> 59 01 4b 46 7f ff 07 10 a2 t=21562
> user@host:/sys/bus/w1/devices $
> ```

Can you add the same output from a shell inside the container?
`docker exec homeassistant ls /sys/bus/w1/devices`

**Comment by Napoleon-BlownApart** [comment]
Thanks your attention.
Here's the output you asked for:
```
user@host:/sys/bus/w1/devices $ docker exec HomeAssistant ls /sys/bus/w1/devices
28-00000588c73f
28-00000589da65
w1_bus_master1
user@host:/sys/bus/w1/devices $
```

**Comment by Napoleon-BlownApart** [comment]
And here is a listing using `ls -lha` instead of `ls`.  It has a different output:

```
user@host:/sys/bus/w1/devices $ docker exec HomeAssistant ls -lha /sys/bus/w1/devices
total 0
drwxr-xr-x    2 root     root           0 Oct  4 12:27 .
drwxr-xr-x    4 root     root           0 Oct  4 12:27 ..
lrwxrwxrwx    1 root     root           0 Oct  4 16:40 00-400000000000 -> ../../../devices/w1_bus_master1/00-400000000000
lrwxrwxrwx    1 root     root           0 Oct  4 16:40 00-800000000000 -> ../../../devices/w1_bus_master1/00-800000000000
lrwxrwxrwx    1 root     root           0 Oct  4 13:17 w1_bus_master1 -> ../../../devices/w1_bus_master1
user@host:/sys/bus/w1/devices $
```

**Comment by thecode** [comment]
Lets see if we something will come up in the diagnostics:
![image](https://user-images.githubusercontent.com/1858925/193751735-bbb13e80-759c-43bd-ba23-27805874a1a9.png)
In this page download the diagnostics and attach in a comment

**Comment by Napoleon-BlownApart** [comment]
[config_entry-onewire_sysbus-782d0515635d299c43fc43cf18d21d6c.json.txt](https://github.com/thecode/ha-onewire-sysbus/files/9703366/config_entry-onewire_sysbus-782d0515635d299c43fc43cf18d21d6c.json.txt)

**Comment by Napoleon-BlownApart** [comment]
> And here is a listing using `ls -lha` instead of `ls`. It has a different output:
> 
> ```
> user@host:/sys/bus/w1/devices $ docker exec HomeAssistant ls -lha /sys/bus/w1/devices
> total 0
> drwxr-xr-x    2 root     root           0 Oct  4 12:27 .
> drwxr-xr-x    4 root     root           0 Oct  4 12:27 ..
> lrwxrwxrwx    1 root     root           0 Oct  4 16:40 00-400000000000 -> ../../../devices/w1_bus_master1/00-400000000000
> lrwxrwxrwx    1 root     root           0 Oct  4 16:40 00-800000000000 -> ../../../devices/w1_bus_master1/00-800000000000
> lrwxrwxrwx    1 root     root           0 Oct  4 13:17 w1_bus_master1 -> ../../../devices/w1_bus_master1
> user@host:/sys/bus/w1/devices $
> ```

Not sure why the ID's are different when using `ls -lha`?

**Comment by Napoleon-BlownApart** [comment]
I configured my `configuration.yaml` file to log debug messages for this integration, and suddenly I have my two thermometers, **plus** two others that don't exist.

```
logger:
  default: info
  logs:
    custom_components.onewire_sysbus: debug
```
![MyThermos_n_more](https://user-images.githubusercontent.com/6958677/193774785-f538e47a-5527-46c6-aa55-a9e87514a93a.png)

I do not have the last 2 mentioned in the card and have no idea how they were detected.

**Comment by Igor2091** [comment]
I have the same problem - i don't see ds20 , but in logs i see

2022-10-04 12:44:05.354 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration onewire_sysbus which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-10-04 12:44:18.105 INFO (SyncWorker_5) [homeassistant.loader] Loaded onewire from homeassistant.components.onewire

**Comment by Napoleon-BlownApart** [comment]
I had a look at the `diagnostics` JSON file, I notice that the devices array has 2 entries for each of my thermometers; one for the `onewire_sysbus`, the other for the `manufacturer`.  It might be that the python code is not dealing with this properly?

```
{
  "home_assistant": {
    "installation_type": "Home Assistant Container",
    "version": "2022.9.7",
    "dev": false,
    "hassio": false,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "armv7l",
    "timezone": "Australia/Brisbane",
    "os_name": "Linux",
    "os_version": "5.10.103-v7l+",
    "run_as_root": true
  },
  "custom_components": {
    "simpleicons": {
      "version": "v2.0.0",
      "requirements": [
        "simpleicons==6.16.0"
      ]
    },
    "nodered": {
      "version": "1.1.2",
      "requirements": []
    },
    "onewire_sysbus": {
      "version": "2022.10.0",
      "requirements": [
        "pi1wire==0.2.0"
      ]
    },
    "bureau_of_meteorology": {
      "version": "1.1.16",
      "requirements": [
        "iso8601"
      ]
    },
    "hacs": {
      "version": "1.27.2",
      "requirements": [
        "aiogithubapi>=22.2.4"
      ]
    }
  },
  "integration_manifest": {
    "domain": "onewire_sysbus",
    "name": "1-Wire SysBus",
    "documentation": "https://github.com/thecode/ha-onewire-sysbus",
    "issue_tracker": "https://github.com/thecode/ha-onewire-sysbus/issues",
    "config_flow": true,
    "requirements": [
      "pi1wire==0.2.0"
    ],
    "codeowners": [
      "@thecode"
    ],
    "iot_class": "local_polling",
    "loggers": [
      "pi1wire"
    ],
    "version": "2022.10.0",
    "is_built_in": false
  },
  "data": {
    "entry": {
      "title": "/sys/bus/w1/devices/",
      "data": {
        "mount_dir": "/sys/bus/w1/devices/"
      }
    },
    "devices": [
      {
        "identifiers": [
          [
            "onewire_sysbus",
            "28-00000588c73f"
          ]
        ],
        "manufacturer": "Maxim Integrated",
        "model": "28",
        "name": "28-00000588c73f"
      },
      {
        "identifiers": [
          [
            "onewire_sysbus",
            "28-00000589da65"
          ]
        ],
        "manufacturer": "Maxim Integrated",
        "model": "28",
        "name": "28-00000589da65"
      }
    ]
  }
}
```

**Comment by Napoleon-BlownApart** [comment]
I'm looking through my logger output (I've enabled `notset` level logging for now), and there are some interesting results;

```
2022-10-05 21:19:46.314 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [2945674376] Sending {"id":16,"type":"result","success":true,"result":[

... other services ...
... your component ...

{"area_id":null,"configuration_url":"homeassistant://hacs","config_entries":["66de48fed249a461e68fc9d0df8ba270"],"connections":[],"disabled_by":null,"entry_type":"service","id":"7407152fadf6dcaab47ac357e452ad9e","identifiers":[["hacs","496755553"]],"manufacturer":"thecode","model":"integration","name_by_user":null,"name":"Raspberry Pi 1-Wire via sysbus","sw_version":null,"hw_version":null,"via_device_id":null},

... onewire_sysbus devices ...

{"area_id":null,"configuration_url":null,"config_entries":["782d0515635d299c43fc43cf18d21d6c"],"connections":[],"disabled_by":null,"entry_type":null,"id":"c1b5d2e953ffcbcb9a2587233d502c69","identifiers":[["onewire_sysbus","28-00000589da65"]],"manufacturer":"Maxim Integrated","model":"28","name_by_user":null,"name":"28-00000589da65","sw_version":null,"hw_version":null,"via_device_id":null},

{"area_id":null,"configuration_url":null,"config_entries":["782d0515635d299c43fc43cf18d21d6c"],"connections":[],"disabled_by":null,"entry_type":null,"id":"a29764a4414b02c65c08f5bd6ad521c8","identifiers":[["onewire_sysbus","28-c00000001a65"]],"manufacturer":"Maxim Integrated","model":"28","name_by_user":null,"name":"28-c00000001a65","sw_version":null,"hw_version":null,"via_device_id":null},

{"area_id":null,"configuration_url":null,"config_entries":["782d0515635d299c43fc43cf18d21d6c"],"connections":[],"disabled_by":null,"entry_type":null,"id":"8951812a7d73ae7dbf9112930aff60d5","identifiers":[["onewire_sysbus","28-200000001a65"]],"manufacturer":"Maxim Integrated","model":"28","name_by_user":null,"name":"28-200000001a65","sw_version":null,"hw_version":null,"via_device_id":null},

{"area_id":null,"configuration_url":null,"config_entries":["782d0515635d299c43fc43cf18d21d6c"],"connections":[],"disabled_by":null,"entry_type":null,"id":"9650e3b326eca6f0dc5c498b7bbfded1","identifiers":[["onewire_sysbus","28-00000588c73f"]],"manufacturer":"Maxim Integrated","model":"28","name_by_user":null,"name":"28-00000588c73f","sw_version":null,"hw_version":null,"via_device_id":null}
]}
```
(I added the spacing & comments to make it easier to read.)

Note that `homeassistant.components.websocket_api.http.connection` is reporting that there are 4 devices.
Not sure how that happens, but only the 1st and 4th (last) are correct.

**Comment by Napoleon-BlownApart** [comment]
I placed a bunch of `_LOGGER.warning` messages in the `OneWireHub` class to see what devices were being discovered, and the outputs were correct up to when the `onewirehub` object initialisation had finished.
So, I decided to try deleting and re-installing the integration.

Once I reinstalled, I got a window that allowed me to add both devices (I didn't get this window at all previously).  I added a new location for the first device.  For the second device, my location wasn't in the drop-down list, so I typed it in (and the filter showed me the location as I was typing it).

So, both thermometers are now accessible in HA and display the correct data except for two things.
1) The Integration widget told me I had 2 devices and 1 entity, and 
2) On the `Devices & Services` page under `Entities`, I could see both devices, however one of them, for some reason, had the wrong `Integration` mentioned:
![Screen Shot 2022-10-06 at 12 04 50 am](https://user-images.githubusercontent.com/6958677/194082087-930d1d32-161d-4fa1-84c1-110d1b58fc5f.png)

Even more interestingly, the `Integration` field updated itself while I was typing this message and is now showing correctly, and the widget is not showing the number of entities correctly as well.

So, it appears that the integration is now working.
*I wonder why it failed the first time, and why there was a delay before the integration field/widget updated to the correct value, given that it was available from the start and the other device showed the correct value from the start.*
